File: www/scripts/simpleGet.py
#!/usr/bin/env python3

# Import necessary CGI modules
import cgi
import html
import datetime

# while True:
#     pass

current_time = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
# Print the required HTTP headers
print("Status: 200 OK")
print("Content-Type: text/html\n")

# Print the HTML content
print("<html>")
print("<head><title>CGI Example</title></head>")
print("<body>")
print("<h1>Python CGI!</h1>")
print(f"<p>Current Time: {current_time}</p>")
# Use cgi module to get any query parameters passed in the URL
form = cgi.FieldStorage()

# Display the values of any query parameters if present
if "name" in form:

    names = form.getlist("name")
    for user_name in names:
        user_name = html.escape(user_name)
        # user_name = html.escape(form.getvalue("name"))
        print(f"<p>Hello, {user_name}!</p>")
else:
    print("<p>No name parameter provided in the query string.</p>")

print("</body>")
print("</html>")

File: www/scripts/simpleGet.pl
#!/usr/bin/env perl

use strict;
use warnings;
use CGI;

# Create CGI object
my $cgi = CGI->new;

# Set content type
print $cgi->header("text/html");

# Get the current time
my ($sec, $min, $hour, $mday, $mon, $year) = localtime();
$year += 1900;
$mon += 1;
my $current_time = sprintf("%04d-%02d-%02d %02d:%02d:%02d", $year, $mon, $mday, $hour, $min, $sec);

# Print the HTML content
print <<END_HTML;
<html>
<head><title>CGI Example</title></head>
<body>
<h1>Perl CGI!</h1>
<p>Current Time: $current_time</p>
END_HTML

# Check if "name" parameter exists
my @names = $cgi->param('name');
if (@names) {
    foreach my $user_name (@names) {
        $user_name = $cgi->escapeHTML($user_name);  # Escape HTML characters
        print "<p>Hello, $user_name!</p>\n";
    }
} else {
    print "<p>No name parameter provided in the query string.</p>\n";
}

print "</body></html>\n";
